### PR TITLE
fix(magicStringTable2): EVAL-358 move magic string in the correct file and replace it

### PR DESCRIPTION
**Pull request by simonchaval**
## Describe your changes
Move magic string in the correct file and make changes in the code.
## Issue ticket number and link
https://entsupport.gdapublic.fr/browse/EVAL-358
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java [hunk 1]
```diff
@@ -787,15 +788,15 @@ public void handle(Either<String, JsonArray> event) {
     public void checkDataOnClasses(String[] idClasses, final Handler<Either<String, JsonArray>> handler) {
         StringBuilder query = new StringBuilder();
         JsonArray values = new fr.wseduc.webutils.collections.JsonArray();
-        query.append(" SELECT devoirs.id, devoirs.name, id_groupe, COUNT(competences_devoirs.id) as nbcompetences ")
+        query.append(" SELECT devoirs.id, devoirs.name, id_groupe, COUNT(" + Field.COMPETENCES_DEVOIRS + ".id) as nbcompetences ")
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java:791]: devoirs

File: src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java [hunk 2]
```diff
@@ -787,15 +788,15 @@ public void handle(Either<String, JsonArray> event) {
...
                 .append(" FROM " + Competences.COMPETENCES_SCHEMA + ".devoirs ")
                 .append(" LEFT JOIN notes.rel_devoirs_groupes ")
                 .append(" ON rel_devoirs_groupes.id_devoir = devoirs.id ")
-                .append(" LEFT OUTER JOIN " + Competences.COMPETENCES_SCHEMA + ".competences_devoirs ")
-                .append(" ON devoirs.id = competences_devoirs.id_devoir ")
+                .append(" LEFT OUTER JOIN " + Competences.COMPETENCES_SCHEMA + "." + Field.COMPETENCES_DEVOIRS)
+                .append(" ON devoirs.id = " + Field.COMPETENCES_DEVOIRS + ".id_devoir ")
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java:796]: devoirs

**Comment by lapeyronnie** [comment]
refait un point globale sur devoirs, competences et competences_notes

**Comment by simonchaval** [comment]
Wrong method, now gonna use file the existant "Field.java" with magic string key to update in the continuity of the project.
cf : [This PR](https://github.com/OPEN-ENT-NG/competences/pull/127)

**Pull request closed without merging**